integration/spark2/src/main/scala/org/apache/carbondata/datamap/IndexDataMapRebuildRDD.scala

```diff
@@ -144,13 +148,113 @@ class OriginalReadSupport(dataTypes: Array[DataType]) extends CarbonReadSupport[
   }
 }
 
+/**
+ * This class will generate row value which is raw bytes for the dimensions.
+ */
+class RawBytesReadSupport(segmentProperties: SegmentProperties, indexColumns: Array[CarbonColumn])
```

Comment:
please add this as Java code in carbon-core module

Reply:
ehh, There is another ReadSupport called 'OriginalReadSupport' in this file which is similar with this 'RawBytesReadSupport'. Do we need to move it too?
Besides, This class is only used IndexRebuildDmRDD, I think it's better to keep them here.

Reply:
besides, currently I can not find a readsupport that in module carbon-core, so better to leave it where it is needed.

![image](https://user-images.githubusercontent.com/10445758/42411659-17c802f2-8232-11e8-8f43-cbb70b305427.png)

Reply:
ok. I think we can refactor it later

Reply:
Add CARBONDATA-2709 to refactor this